src/states/away/CountdownToAway.java

```diff
@@ -1,7 +1,69 @@
 package states.away;
 
+import events.TimerRanOut;
+import events.TimerTicked;
+import events.UncheckZone;
 import states.Countdown;
+import states.SecurityContext;
 
 public class CountdownToAway extends Countdown {
+	private static CountdownToAway instance;
 
+	/**
+	 * Private for the singleton pattern
+	 */
+	private CountdownToAway() {
+	}
+
+	/**
+	 * For singleton
+	 * 
+	 * @return the object
+	 */
+	public static CountdownToAway instance() {
+		if (instance == null) {
+			instance = new CountdownToAway();
+		}
+		return instance;
+	}
+
+	/**
+	 * Processes a timer tick, generates a Timer Ticked event
+	 */
+	@Override
+	public void handleEvent(TimerTicked event) {
+
+		SecurityContext.instance().showSecondsToAway(timer.getTimeValue());
+	}
+
+	/**
+	 * Processes a timer tick, generates a Timer Ran Out event
+	 */
+	@Override
+	public void handleEvent(TimerRanOut event) {
+		SecurityContext.instance().showSecondsToAway(0);
+		SecurityContext.instance().changeState(ArmedAway.instance());
+	}
+
+	/**
+	 * Processes one zone being unchecked
+	 */
+	@Override
+	public void handleEvent(UncheckZone event) {
+
+		SecurityContext.instance().changeState(CountdownToAwayNotReady.instance());
+	}
+
+	@Override
+	public void enter() {
+
+		SecurityContext.instance().showSecondsToAway(timer.getTimeValue());
+	}
+
+	@Override
+	public void leave() {
+
+		super.leave();
+		SecurityContext.instance().showAway();
```

Comment:
I think it should showAway, as it enters Away, not as it leaves countdown.

Reply:
Very true, I will adapt it